cumulusci/cli/ui.py

```diff
@@ -133,3 +133,128 @@ def _dim_value(self, val):
         # If a string has been wrapped, wrap each line to avoid dimming table borders
         dimmed_strs = [click.style(line, dim=True) for line in val.split("\n")]
         return "\n".join(dimmed_strs)
+
+
+def _soql_table(results, truncated):
+    if results:
+        if truncated:
+            assert results[-1] == truncated
+            first_row = results[0]
+            fake_row = {k: "" for k, v in first_row.items()}
+            first_column = list(first_row)[0]
+            fake_row[first_column] = truncated
+
+            results[-1] = fake_row
+
+        headings = list(results[0].keys())
+        return str(
+            CliTable(
+                [headings] + [list(map(str, r.values())) for r in results],
+                wrap_cols=headings,
+            )
+        )
+    else:
+        return str(CliTable([["No results"]]))
+
+
+def _summarize(field):
+    if field["referenceTo"]:
+        allowed = field["referenceTo"]
+    elif field["picklistValues"]:
+        allowed = [value["value"] for value in field["picklistValues"]]
+    else:
+        allowed = field["type"]
+    return (field["name"], allowed)
+
+
+class SimpleSalesforceUIHelpers:
+    def __init__(self, sf):
+        self._sf = sf
+
+    def query(self, query, format="table", include_deleted=False, max_rows=100):
+        """Return the result of a Salesforce SOQL query.
+
+        Arguments:
+
+            * query -- the SOQL query to send to Salesforce, e.g.
+                    SELECT Id FROM Lead WHERE Email = "waldo@somewhere.com"
+            * include_deleted -- True if deleted records should be included
+            * format -- one of these values:
+                - "table" -- printable ASCII table (the default)
+                - "obj" -- ordinary Python objects
+                - "pprint" -- string in easily readable Python dict shape
+                - "json" -- JSON
+            * max_rows -- maximum rows to output, defaults to 100
+
+        For example:
+            query("select Name, Id from Account", format="pprint")
+            contact_ids = query("select count(Id) from Contact", format="obj")
+        """
+        results = self._sf.query_all(query, include_deleted=include_deleted)["records"]
```

Comment:
Now we're getting everything even if we're going to throw away the results beyond max_rows. The more careful approach would be to go one batch at a time using query_more and return once max_rows is reached. But I'm okay with this as a first pass.

Reply:
Yeah, it's just for quick interactive uses, not to replace simple_salesforce itself. The max_rows feature overlaps with the LIMIT feature in any case and is primarily just so you don't accidentally dump a million rows on your terminal if you forgot which org you were talking to.